stable/concourse/templates/worker-statefulset.yaml

```diff
@@ -56,9 +59,12 @@ spec:
             preStop:
               exec:
                 command:
-                  - "/bin/sh"
-                  - "-c"
-                  - "concourse retire-worker --name=${HOSTNAME}"
+                  - /bin/sh
+                  - -c
+                  - |-
+                    while ! concourse retire-worker --name=${HOSTNAME} | grep -q worker-not-found; do
```

Comment:
Is there a condition where retire-worker will never come back good and this will be stuck in the loop?

Reply:
Yes, I've seen this happen where ATC can't/won't clean up the worker as a result of calling `retire-worker`. On termination the terminationGracePeriod will take affect and the container will be killed. When it comes back it could get stuck in this loop. The pod will be live, but the worker won't register because the worker process hasn't started yet, it's still trying to retire-worker because the old worker still exists in concourse. 

The only resolution is to manually intervene in the fly cli with `fly prune-worker`. I've requested that the `concourse` command add the option to forcefully delete: https://github.com/concourse/concourse/issues/1457#issuecomment-353766379, which would be called in the startup script instead of looping over retire-worker. 

This deserves a paragraph in the readme, but I/m' not sure if we can make this any better at the moment, unless we can find another way to send that delete request.

Reply:
Actually, there is a way to make this better, have the liveness probe ensure the concourse process is up in addition to checking for fatal errors, and have the livenessProbeDelay tuneable.